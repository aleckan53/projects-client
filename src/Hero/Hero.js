import React from 'react'
import './Hero.css'
import { Link } from 'react-router-dom' 

import {IoMdCreate} from 'react-icons/io'

const Hero = props => {
  const details = props.owner || props.project
  return <header className="Hero"
    style={{
      backgroundPosition: "center",
      backgroundSize: `cover`,
      backgroundRepeat: "no-repeat",
      backgroundImage: `url(${details.img || details.avatar})`
    }}>
    <h2>{details.title || details.firstName + " "+ details.lastName}</h2>
    {!props.editable ? '' : <Link to={`/project/${props.projectId}/edit`} className="edit-link"><IoMdCreate/></Link>}
    {!props.owner ? '' : <div className="edit-profile"><IoMdCreate/></div>}
  </header>

}

export default Hero