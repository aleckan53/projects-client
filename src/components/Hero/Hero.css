.Hero{
  box-sizing: border-box;
  position: relative;
  width: 100%;
  height: 30vh;
  background-position: center;
  background-size: cover;
  background-repeat: no-repeat;

}

.Hero h1 {
  padding: 0 5%;
  position: absolute;
  z-index: 3;
  bottom: 3%;
  color: white;
  font-size: 1.5rem;
}

.edit-link, .edit-profile{
  padding: 0 5%;
  background: none;
  border: none;
  position: absolute;
  right: 0;
  top: 5%;
}

.edit-link svg, .edit-profile svg {
  fill: rgba(169,169,169,0.7);
  height: 2rem;
  width: 2rem;
}

.edit-link svg:hover, .edit-profile svg:hover {
  fill: var(--main-color)
}