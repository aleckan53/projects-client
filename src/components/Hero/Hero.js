import React from 'react'
import styles from './Hero.module.css'
import { Link } from 'react-router-dom'
import { IoMdSettings as Icon } from 'react-icons/io' 
import { HeaderBtn } from '../Basic/Basic'

const Hero = props => {

  const backgroundImgStyle = {
    backgroundImage: `url(${props.img})`
  }

  return (
    <header 
      className={`${styles.hero}`}
      style={backgroundImgStyle}>
      <h1 className={styles.h1}>{props.heading}</h1>
      { !props.show ? '' :
        <Link to={`/projects/${props.id}/edit`}>
          <HeaderBtn 
            icon={Icon}
            className={styles.editBtn}/>
        </Link>
      }
      {props.children}
    </header>
  )
}

export default Hero