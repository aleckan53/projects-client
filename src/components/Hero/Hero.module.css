.hero {
  box-sizing: border-box;
  position: relative;
  width: 100%;
  height: 30vh;
  background-position: center;
  background-size: cover;
  background-repeat: no-repeat;
  padding: var(--padding-side)
}

.hero .h1 {
  position: absolute;
  z-index: 3;
  bottom: 3%;
  color: var(--sub-text);
  font-size: 1.5rem;
}

.editBtn {
  color: var(--main-text-30)!important;
  position: absolute;
  right: 0;
  padding: var(--padding-side);
  padding-top: 5%;
}

.editBtn:hover {
  color: var(--main-color)!important;
}

