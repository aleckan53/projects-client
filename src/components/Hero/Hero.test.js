import React from 'react'
import ReactDOM from 'react-dom'
import Hero from './Hero'
import renderer from 'react-test-renderer'

describe('Hero component', ()=> {
  it('renders without crashing', ()=> {
    const div = document.createElement('div')
    ReactDOM.render(<Hero />, div)
    ReactDOM.unmountComponentAtNode(div)
  })
  
  it('renders UI as  expected', ()=> {
    const tree = renderer
      .create(
      <Hero
        img="https://carepharmaceuticals.com.au/wp-content/uploads/sites/19/2018/02/placeholder-600x400.png"
        heading="test"/>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})