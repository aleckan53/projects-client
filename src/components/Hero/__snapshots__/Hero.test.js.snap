// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero component renders UI as  expected 1`] = `
<header
  className="hero"
  style={
    Object {
      "backgroundImage": "url(https://carepharmaceuticals.com.au/wp-content/uploads/sites/19/2018/02/placeholder-600x400.png)",
    }
  }
>
  <h1
    className="h1"
  >
    test
  </h1>
</header>
`;
